core/src/main/java/org/apache/carbondata/core/scan/collector/ResultCollectorFactory.java

```diff
@@ -45,31 +46,37 @@
    * @return
    */
   public static AbstractScannedResultCollector getScannedResultCollector(
-      BlockExecutionInfo blockExecutionInfo) {
+      BlockExecutionInfo blockExecutionInfo, QueryStatisticsModel queryStatisticsModel) {
```

Comment:
Instead of changing constructor, can we add statistics model to block execution info??

Reply:
I think it is better to pass in constructor instead of making the blockExecutionInfo object more heavier. Also because the QueryStatisticsModel is at the reader level and one reader can have multiple blocks so in my opinion keeping in blockExecutionInfo will not be a good idea.